skietbaan_fe/src/components/Login.js

```diff
@@ -1,98 +1,135 @@
+      var re = /(?=.*\d)(?=.*[a-z]).{6,}/;
+      return re.test(str);
+    }
+
+    function validateEmail(email) {
+      var re = /.+@.+\..+/;
+      return re.test(String(email).toLowerCase());
+    }
+
+    function validateUsername(username) {
+      var re = /[a-zA-Z]/;
+        return !re.test(String(username));
+    }
+    
 class App extends Component {
   constructor(props){
     super(props);
     this.state = {
       usernameValue : "",
       emailValue : "",
       passwordValue:"",
-      invalid:false
+      invalidPassword:false,
+      invalidEmail:false,
+      invalidUsername: false,
+      validForm:false
     }
-    this.showAlert = this.showAlert.bind(this);
+    this.Login = this.Login.bind(this);
     this.handleChange = this.handleChange.bind(this);
-    this.postData = this.handleChange.bind(this);
   }
   handleChange({ target }) {
     this.setState({
-      [target.name]: target.value
+      [target.name]: target.value,
+      invalidPassword:false,
+      invalidEmail: false,
+      invalidUsername: false
     });
-  }
-   postData(url = ``, data = {}) {
-    // Default options are marked with *
-      return fetch(url, {
-          method: "POST", // *GET, POST, PUT, DELETE, etc.
-          mode: "cors", // no-cors, cors, *same-origin
-          cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
-          credentials: "same-origin", // include, *same-origin, omit
-          headers: {
-              "Content-Type": "application/json",
-              // "Content-Type": "application/x-www-form-urlencoded",
-          },
-          redirect: "follow", // manual, *follow, error
-          referrer: "no-referrer", // no-referrer, *client
-          body: JSON.stringify(data), // body data type must match "Content-Type" header
-      })
-      .then(response => response.json()); // parses response to JSON
-  }
-  
-  showAlert() {
-
 
-    if(this.state.passwordValue === "" || this.state.passwordValue === null)
+    var Valid = true;
+    if(!validatePassword(this.state.passwordValue))
+    {
+        this.setState({
+          invalidPassword : true
+        });
+        Valid = false;
+    }
+    if(!validateEmail(this.state.emailValue))
+    {
+        this.setState({
+          invalidEmail : true
+        });
+        Valid = false;
+    }
+    if(validateUsername(this.state.usernameValue))
+    {
+        this.setState({
+          invalidUsername : true
+        });
+        Valid = false;
+    }
     this.setState({
-      invalid : true
+      validForm:Valid
     });
-
-
+  }
+  
+  Login() {
+    if(this.state.validForm)
+    {
     var RequestObject = {
-        "Username":this.state.usernameValue,
-        "Password":this.state.emailValue,
-        "Surname":this.state.passwordValue
+      "Username":this.state.usernameValue,
+      "Password":this.state.emailValue,
+      "Surname":this.state.passwordValue
     }
     console.log(RequestObject);
```

Comment:
Remove this console.log.
It will be used against you by malicious people if it goes to production.

Reply:
removed